### PR TITLE
[ci] Retry clones and fetches

### PR DESCRIPTION
**Pull request by danking**
This got a bit complicated to ensure idempotence.

This should make issues related to GitHub hiccups during clones and fetches even
rarer than they are now (retry three times, as we do for uploading).

I added a function called `clone` which:
- creates a temporary,
- clones into the temporary directory, and
- copies every file in the directory (including hidden files) into the current
  directory

I added retries to clones and fetches (which are already idempotent).

I removed duplication of cloning and fetching scripts. I also moved the shell retry function to `hailtop.utils`.

### CODE REVIEW

File: ci/ci/constants.py [hunk 1]
```diff
@@ -21,3 +21,6 @@
     'gsarma',
     'mkveerapen'
 }
+
```
Comment (jigold) [inline ci/ci/constants.py]: Delete

File: ci/ci/github.py [hunk 2]
```diff
@@ -132,6 +132,30 @@ def __init__(self, exception, attributes=None):
...
 
+def clone_or_fetch_script(repo):
+    return """
+{ RETRY_FUNCTION_SCRIPT }
+
+function clone() {{
+    dir=$(mktemp -d)
+    git clone {shq(repo)} $dir
+    for x in $(ls -A $dir); do
+        if [ -e "$dir/$x" ]; then mv -- "$dir/$x" ./; fi
```
Comment (jigold) [inline ci/ci/github.py]: Why do you need to check the existence of `$dir/$x`? Shouldn't it exist since you list the directory?

Comment (jigold) [inline ci/ci/github.py]: What is the `--` for in the mv command?

Comment (danking) [reply ci/ci/github.py]: It indicates to `mv` that the next arguments are the sources and target. If a directory was named `-x` then, without `--`, `mv` would think you sent it the `-x` argument.

Comment (danking) [reply ci/ci/github.py]: I copied this from stack overflow and honestly am not sure. I'll remove it.

**Pull request merged**